### PR TITLE
Add CONTRIBUTING.md

### LINKED ISSUE #12: Add CONTRIBUTING.md
Add CONTRIBUTING.md with rules for contributors in their pull requests

### PR DESCRIPTION
**Pull request by bleudev**
closes #12

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# 📖 Contributing rules in `ufpy` repository
+
+If you want to contribute, you have to read this rules and follow they.
```
Comment (TheHakerTech) [inline CONTRIBUTING.md:3]: If you want to contribute **the project**, you **must** read this rules and follow they.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## 💻 Code style
+
+- Formatter: `black`
+- Optimizer of imports: `isort`
+- Linter: `Pylint`
+- Generic style: `True`
+
+Generic style is style where are using `typing.Generic` class in all `useful classes`
```
Comment (TheHakerTech) [inline CONTRIBUTING.md:12]: Use `typing.Generic` like generic style in `useful classes`.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+T = TypeVar('T')
+
+class A(Generic[T]): 
+    def __init__(self, a: T): ...
+```
+
+- `@overload`s: `True`
+
+You must define `@typing.overload`s in public methods when your method has two and more ways to define arguments
```
Comment (TheHakerTech) [inline CONTRIBUTING.md:26]: **Use decorator** `@typing.overload`s in public methods **if** your method has two and more **variants** to define arguments

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+@overload
+def a(x: list): ...
+@overload
+def a(x: dict): ...
+def a(x: int | list | dict): ...
+```
+
+- `__all__` variable: `True`
+
+`__all__` variable is variable with names of every class,
```
Comment (TheHakerTech) [inline CONTRIBUTING.md:40]: `__all__` is a dict with names of every class,

**Review by TheHakerTech — COMMENTED** [review]
good

File: CONTRIBUTING.md (hunk 4 above)
Comment (bleudev) [reply CONTRIBUTING.md:40]: > `__all__` is a dict with names of every class,

it's a tuple)

**Review by bleudev — COMMENTED** [review]
Approved!

**Pull request merged**